flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/HITSExample.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.graph.example;
+
+import org.apache.flink.api.common.ProgramDescription;
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.graph.Edge;
+import org.apache.flink.graph.Graph;
+import org.apache.flink.graph.Vertex;
+import org.apache.flink.graph.library.HITS;
+import org.apache.flink.graph.utils.Hits;
+import org.apache.flink.graph.utils.Tuple3ToEdgeMap;
+import org.apache.flink.util.Collector;
+
+/**
+ * This program is an example for HITS algorithm.
+ * the result is either a hub value or authority value base user selection.
+ *
+ * If no arguments are provided, the example runs with a random graph of 10 vertices
+ * and random edge weights.
+ */
+
+public class HITSExample implements ProgramDescription {
+    @SuppressWarnings("serial")
+    public static void main(String[] args) throws Exception {
+
+        if(!parseParameters(args)) {
+            return;
+        }
+
+        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+        DataSet<Edge<Long, String>> links = getEdgesDataSet(env);
+
+        Graph<Long, Double, String> network = Graph.fromDataSet(links, new MapFunction<Long, Double>() {
+
+            public Double map(Long value) throws Exception {
+                return 1.0;
+            }
+        }, env);
+
+    // add  graph to HITS class with iteration value and hub or authority enum value.
+        DataSet<Vertex<Long, Double>> HitsValue =network.run(
+                    new HITS<Long>(Hits.AUTHORITY,maxIterations)).getVertices();
+
+        if (fileOutput) {
+            HitsValue.writeAsCsv(outputPath, "\n", "\t");
+        } else {
+            HitsValue.print();
+        }
+//        env.execute("HITS algorithm");
```

Comment:
The env.execute() caused problems for you because you were only testing print(). You don't need to do an env.execute() on the else branch as print() now has the same status as count() and collect(), but this command is still needed on the then branch(i.e. after writeAsCsv).